presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -605,9 +609,24 @@ public static String computeBucketedFileName(String queryId, int bucket)
     public static String getFileExtension(JobConf conf, StorageFormat storageFormat)
     {
         // text format files must have the correct extension when compressed
-        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT) || !HiveIgnoreKeyTextOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
+        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT)) {
```

Comment:
Can we move this if statement after checking for other storage format ?

Reply:
U mean to add the above line after last if check?
(HiveSequenceFileOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {

Reply:
@Praveen2112 @shandeep00 : should this be a property of HiveStorageFormat?